packages/reactstrap-validation-date/src/AvDate.js

```diff
@@ -0,0 +1,217 @@
+    const { value, format } = this.state;
+    const date = moment(
+      value,
+      [isoDateFormat, format, 'MMDDYYYY', 'YYYYMMDD'],
+      true
+    );
+    if (date.isValid()) return date;
+    return null;
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+
+    if (props.type.toLowerCase() === 'date' && inputType.date) {
+      this.state.format = isoDateFormat;
+    } else {
+      this.state.format =
+        (props.validate && props.validate.date && props.validate.date.format) ||
+        'MM/DD/YYYY';
+    }
+
+    this.state.focused = false;
+  }
+
+  onFocusChange = ({ focused }) => {
+    const { onPickerFocusChange, name } = this.props;
+    const touched = this.context.FormCtrl.isTouched(name);
+    if (!touched && !focused) {
+      this.context.FormCtrl.setTouched(name);
+    }
+
+    this.setState({ focused });
+
+    if (onPickerFocusChange) onPickerFocusChange({ focused });
+  };
+
+  onPickerChange = async value => {
+    const { format } = this.state;
+    const { name, onChange } = this.props;
+    if (value === null) return;
+
+    let val = value;
+    if (val instanceof Object && val.isValid()) {
+      val = val.format(format);
+    }
+
+    this.context.FormCtrl.getInput(name)
+      .getValidatorProps()
+      .onChange(val);
+
+    this.setState({ value: val }, () => {
+      if (onChange) onChange(val);
+    });
+  };
+
+  onClose = ({ date }) => {
+    const { format } = this.state;
+
+    const { onBlur } = this.context.FormCtrl.getInput(
+      this.props.name
+    ).getValidatorProps();
+
+    onBlur(date && date.format(format));
+  };
+
+  valueParser = value => {
+    if (this.state.format === isoDateFormat) return value;
+    const date = moment(
+      value,
+      [this.state.format, 'MMDDYYYY', 'YYYYMMDD'],
+      true
+    );
+    if (date.isValid()) return date.format(isoDateFormat);
+    return value;
+  };
+
+  valueFormatter = value => {
+    const date = moment(
+      value,
+      [isoDateFormat, this.state.format, 'MMDDYYYY', 'YYYYMMDD'],
+      true
+    );
+    if (date.isValid()) return date.format(this.state.format);
+    return value;
+  };
+
+  render() {
+    const {
+      datepicker,
+      calendarIcon,
+      hideIcon,
+      className,
+      datePickerProps,
+      min,
+      max,
+      type,
+      name,
+      ...attributes
+    } = this.props;
+
+    const pickerId = `${(this.props.id || name).replace(
+      /[^a-zA-Z0-9]/gi,
+      ''
+    )}-btn`;
+
+    const touched = this.context.FormCtrl.isTouched(name);
+    const hasError = this.context.FormCtrl.hasError(name);
+
+    const classes = classNames(
+      className,
+      touched ? 'is-touched' : 'is-untouched',
+      this.context.FormCtrl.isDirty(name) ? 'is-dirty' : 'is-pristine',
+      this.context.FormCtrl.isBad(name) ? 'is-bad-input' : null,
+      hasError ? 'av-invalid' : 'av-valid',
+      touched && hasError && 'is-invalid',
+      !this.state.value && 'current-day-hightlight'
```

Comment:
i know that's how it's defined in css file but i'm concerned someone sees that as a typo and ~fixes~ breaks it

Reply:
Yeah that needs to be fixed. typo. If there's a better way for optional highlighting the today's date I'm for it.